lib/config/ldaptlsmode.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package config
+
+type LDAPTLSMode int
+
+const (
+	LDAPTLSModePlain LDAPTLSMode = iota // default is notls
+	LDAPTLSModeTLS
+	LDAPTLSModeStartTLS
+)
+
+func (t LDAPTLSMode) String() string {
+	switch t {
+	case LDAPTLSModePlain:
+		return "plain"
```

Comment:
Now that I think about it, it's not plain tls, it's simply not tls at all, so perhaps this should not be LDAPTLSMode, but LDAPTransport or LDAPProtocol or something alike. Or the mode should be none, (as in non-tls).

Sorry for distracting you here earlier.

Reply:
Lets do LDAPTransport with LDAPTransportPlain, LDAPTransportTLS and LDAPTransportStartTLS?